### PR TITLE
Add new method for retrieving range of codespan

### PR DESCRIPTION
**Pull request by fnkabit**
This would be required in order to fix https://github.com/pbek/QOwnNotes/issues/1967

### CODE REVIEW

**Comment by fnkabit** [comment]
@pbek I think we need to restart the CI build.

**Comment by Waqar144** [comment]
Can `findPositionInRanges()` not be used?

File: markdownhighlighter.cpp [hunk 1]
```diff
@@ -2196,6 +2196,23 @@ bool MarkdownHighlighter::isPosInACodeSpan(int blockNumber, int position) const
...
+    const QVector<InlineRange> rangeList = _ranges.value(blockNumber);
+    const auto it = std::find_if(rangeList.cbegin(), rangeList.cend(),
+                                     [position](const InlineRange& range){
+        if (position > range.begin && position < range.end && range.type == RangeType::CodeSpan)
+            return true;
+        return false;
+    });
+
+    if (it == rangeList.cend()) {
+        return {};
```
Comment (Waqar144) [inline markdownhighlighter.cpp:2210]: Should this not return - 1, - 1 instead?

**Comment by Waqar144** [comment]
Also, `isPosInACodeSpan` and this method is almost the same, so should we not merge them? Or better yet, make them a bit more generic, like `getRange(type, block, pos)`

Thoughts?

**Comment by fnkabit** [comment]
> Also, `isPosInACodeSpan` and this method is almost the same, so should we not merge them? Or better yet, make them a bit more generic, like `getRange(type, block, pos)`
> 
> Thoughts?

Updated

**Comment by pbek** [comment]
Oh great, Qt seems to have removed all their repos below 5.10 or so. 😞  https://github.com/pbek/QOwnNotes/ is not affected yet because the files are still in cache, but soon the builds will break too. Oh man...

File: markdownhighlighter.cpp (hunk 1 above)
Comment (pbek) [reply markdownhighlighter.cpp]: @fnkabit ?

**Comment by Waqar144** [comment]
> Oh great, Qt seems to have removed all their repos below 5.9 or so. 😞 https://github.com/pbek/QOwnNotes/ is not affected yet because the files are still in cache, but soon the builds will break too. Oh man...

Guess we have to rely on the old style OBS now

**Comment by pbek** [comment]
> > Oh great, Qt seems to have removed all their repos below 5.9 or so. disappointed https://github.com/pbek/QOwnNotes/ is not affected yet because the files are still in cache, but soon the builds will break too. Oh man...
> 
> Guess we have to rely on the old style OBS now

But then it's already too late... 😬 
I opened https://github.com/jurplel/install-qt-action/issues/66.

File: markdownhighlighter.cpp (hunk 1 above)
Comment (pbek) [reply markdownhighlighter.cpp]: this isn't resolved yet, right?

**Comment by Waqar144** [comment]
> > > Oh great, Qt seems to have removed all their repos below 5.9 or so. disappointed https://github.com/pbek/QOwnNotes/ is not affected yet because the files are still in cache, but soon the builds will break too. Oh man...
> > 
> > 
> > Guess we have to rely on the old style OBS now
> 
> But then it's already too late... grimacing
> I opened [jurplel/install-qt-action#66](https://github.com/jurplel/install-qt-action/issues/66).

Qt 5.3 / 5.6 are too old, so I guess they just dropped them

**Comment by pbek** [comment]
> Qt 5.3 / 5.6 are too old, so I guess they just dropped them

they dropped everything up to 5.8

**Comment by Waqar144** [comment]
> > Qt 5.3 / 5.6 are too old, so I guess they just dropped them
> 
> they dropped everything up to 5.8

5.9 is also deprectated, but i think they will keep it for another year or so.

**Pull request merged**